### PR TITLE
pin window left, right or centered

### PR DESCRIPTION
**Pull request by BakeRolls**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -140,8 +140,25 @@ module.exports = function create (opts) {
...
+        if (opts.pin === 'left') {
+          x = Math.floor(trayPos.x)
+        } else if (opts.pin === 'right') {
+          x = Math.floor(trayPos.x - (opts.width || 200) + trayPos.width)
+        } else {
+          x = Math.floor(trayPos.x - ((opts.width / 2) || 200) + (trayPos.width / 2))
+        }
+      }
+
+      if (x < 0) {
```
Comment (jenslind) [inline index.js]: This seems to remove the support for multiple screens.

**Comment by BakeRolls** [comment]
I don't have a second monitor until for the next days. I'll test it later.

File: index.js (hunk 1 above)
Comment (BakeRolls) [inline index.js]: I don't have a second monitor until for the next days. I'll test it later.

**Comment by jenslind** [comment]
Also to make it cleaner and more elegant. :cat2:  We could use a object literal instead of all the if, else.

Something like this:
```Javascript
var pin = {
  left: Math.floor(trayPos.x),
  right: Math.floor(trayPos.x - (opts.width || 200) + trayPos.width),
  center: Math.floor(trayPos.x - ((opts.width / 2) || 200) + (trayPos.width / 2))
}

var x = (opts.x !== undefined) ? opts.x : (pin[opts.pin] || pin['center'])
```

:beers:

**Comment by maxogden** [comment]
I'm not opposed to this feature, but maybe we should think about more cleanly separating the window positioning logic, perhaps even into a separate module?

**Comment by jenslind** [comment]
Yeah, maybe not a bad idea to make positioning go into its own module. A module like that will probably be helpful in other electron projects.

**Comment by jenslind** [comment]
I have started to work on a module for this now. https://github.com/jenslind/electron-positioner
Not sure if `positioner` is a valid word. :dancer:

**Comment by jenslind** [comment]
See https://github.com/maxogden/menubar/pull/54

**Pull request closed without merging**